### ISSUE TITLE
Determine order of operations for spawned code

### ISSUE THREAD

**Issue by cowboyd** [thefrontside/effection#835]
Currently when executing, spawned operations begin executing before control is returned to the spawner. This means that if you want to do things relevant to the caller, then you need to have explicit logic in the spawned process _not_ to proceed until the caller is ready. For example, this should probably work, but it doesn't.

```ts
let { send, ...messages } = createChannel<string, never>();

yield* spawn(function*() {
  yield* send("hello");
});

yield* spawn(function*() {
  yield* send("world");
});

for (let message of yield* each(messages)) {
  console.log(message);
  yield* each.next();
}
```

In fact, the "hello" and "world" messages will be missed. Instead, you would have to write this as:

```ts
let { send, ...messages } = createChannel<string, never>();

yield* spawn(function*() {
  for (let message of yield* each(messages)) {
    console.log(message);
    yield* each.next();
  }
});

yield* spawn(function*() {
  yield* send("hello");
});

yield* spawn(function*() {
  yield* send("world");
});

yield* suspend();
```

Which is counter-intuitive, especially the `suspend()` operation at the end. The only other option is to have an explicit synchronization mechanism at the beginning of each producer.

```ts
let { send, ...messages } = createChannel<string, never>();

let ready = new Deferred(); // <- this class doesn't actually exist. You'd have to write it.

yield* spawn(function*() {
  yield* ready;
  yield* send("hello");
});

yield* spawn(function*() {
  yield* ready;
  yield* send("world");
});

yield* ready.resolve();

for (let message of yield* each(messages)) {
  console.log(message);
  yield* each.next();
}
```

That's quite awkward. As a result, I think that fine-grained control should ultimately be with the caller such that only once it explicitly yields control, then other operations can run. The kinds of  co-coordination between parents and their children in these examples work against overall composability.  It's better to have the parent be able to do all the sequencing, that way children can just do their thing.

**Comment by cowboyd** [comment]
I think we're going to roll with this as-is.
